mlflow/recipes/steps/automl/flaml.py

```diff
@@ -138,6 +140,10 @@ def _create_model_automl(
     try:
         if primary_metric in _MLFLOW_TO_FLAML_METRICS and primary_metric in evaluation_metrics:
             metric = _MLFLOW_TO_FLAML_METRICS[primary_metric]
+            if metric in _MULTICLASS_TO_FLAML_METRICS and extended_task in [
+                "classification/multiclass"
+            ]:
+                metric = _MULTICLASS_TO_FLAML_METRICS[metric]
```

Comment:
I don' think it's necessary to introduce `_MULTICLASS_TO_FLAML_METRICS` since there's only 1 item required for special handling. `roc_auc` is already defined in `_MLFLOW_TO_FLAML_METRICS`. You can probably do something like
```
if primary_metric == `roc_auc` and extended_task == "classification/multiclass":
  metric = roc_auc_ovr
```